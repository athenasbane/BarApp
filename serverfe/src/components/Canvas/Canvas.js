import React from 'react';
import {
    makeStyles,
    Grid
} from '@material-ui/core';
import socketIOClient from 'socket.io-client';
import { backendURL } from '../../constants';
import Order from '../Order/Order';
const socket = socketIOClient(backendURL);

const useStyles = makeStyles({
    root: {
        width: '100vw',
        height: '100vh',
        backgroundColor: 'black',
        overflow: 'scroll'
    }
});

const Canvas = () => {
    const [ orderData, setOrderData ] = React.useState([]);
    
    const getData = async () => {
        try {
            let response = await fetch(backendURL + '/order');
            let data = await response.json();
            setOrderData(data);
        } catch (e) {
            console.log(e)
        } 
    };

    React.useEffect(() => {
        socket.on('NewOrder', data => {
            setOrderData((oldState = []) => {
                setOrderData([...oldState, data]);
            });
        });
    }, []);

    React.useEffect(() => {
        getData();
    }, []);

    const deliveredHandler = async (id) => {
        await fetch(backendURL + `/order/${id}`, {
            method: 'PATCH',
        });
        await getData();
    };

    const orders = orderData.filter(order => {
        if (order.delivered) { 
            return false;
        }
        return true;
        }).map(order => (<Order order={order} 
                    key={order._id} 
                    deliveredHandler={deliveredHandler}/>)
        );

    const classes = useStyles();
    return (
        <div className={classes.root}>
            <Grid container direction="column" spacing={1}>
                {orders}
            </Grid>
        </div>
    );
}

export default Canvas;