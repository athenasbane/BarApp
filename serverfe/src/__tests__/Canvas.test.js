import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Canvas from '../components/Canvas/Canvas';

describe('Canvas Tests', () => {
    test('renders correctly', () => {
        const tree = renderer
            .create(<Canvas />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
}) 