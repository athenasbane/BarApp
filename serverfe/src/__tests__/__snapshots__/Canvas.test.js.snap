// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas Tests renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-direction-xs-column"
  />
</div>
`;
